lib/runtime/wazero/imports.go

```diff
@@ -1214,9 +1362,30 @@ func ext_default_child_storage_root_version_1(
 
 //export ext_default_child_storage_root_version_2
 func ext_default_child_storage_root_version_2(ctx context.Context, m api.Module, childStorageKey uint64,
-	stateVersion uint32) (ptrSize uint64) {
-	// TODO: Implement this after we have storage trie version 1 implemented #2418
-	return ext_default_child_storage_root_version_1(ctx, m, childStorageKey)
+	stateVersion uint32) (ptrSize uint64) { //skipcq: RVV-B0012
+	rtCtx := ctx.Value(runtimeContextKey).(*runtime.Context)
+	if rtCtx == nil {
+		panic("nil runtime context")
+	}
+	storage := rtCtx.Storage
+	child, err := storage.GetChild(read(m, childStorageKey))
+	if err != nil {
+		logger.Errorf("failed to retrieve child: %s", err)
+		return 0
```

Comment:
Since `ext_default_child_storage_root_version_2` caller expects a `Vec<u8>` I would suggest that in failure cases return a pointer to an empty `Vec<u8>` instead of 0

https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/primitives/io/src/lib.rs#L537